### PR TITLE
doc: add virtual_device_pool block documentation

### PR DESCRIPTION
**Pull request by rbino**
Signed-off-by: Riccardo Binetti <riccardo.binetti@ispirata.com>

### CODE REVIEW

File: guides/blocks/0119-virtual_device_pool.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+* type: consumer
+* input: any kind of message conforming to the [input format](#accepted-input-messages)
+
+This is a consumer block that sends incoming messages to Astarte, acting as one or more virtual
+devices.
+
+Information about Device ID, interfaces and paths are deduced from the message key, so it's
+important that the key conforms to the [supported format](#accepted-input-messages). You can use a
+[lua_mapper block]() to add the appropriate key to your message.
```
Comment (bettio) [inline guides/blocks/0119-virtual_device_pool.md:11]: missing link here `lua_map: 0112-lua_map.html`

File: guides/blocks/0119-virtual_device_pool.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+## `interfaces`
+
+The interfaces supported by the target devices. Note that this should be an array of objects, the
+whole interface definition in must be present for every interface. Right now only datastream
+interfaces are supported.
+
+This is an example of an `interfaces` array:
+
+```json
```
Comment (bettio) [inline guides/blocks/0119-virtual_device_pool.md:121]: sounds weird, but let's try using javascript

File: guides/blocks/0119-virtual_device_pool.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+An object used to configure the target devices. The object _must_ contain these keys:
+
+* `realm`: the realm name (required, string)
+* `jwt`: a JWT with the correct claims to access the target Astarte Pairing API instance. (required,
+  string)
+
+This is an example of a `realms` object:
+
+```
```
Comment (bettio) [inline guides/blocks/0119-virtual_device_pool.md:106]: sounds weird, but let's try using javascript

File: guides/blocks/0119-virtual_device_pool.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+* `realm`: the target Realm for the device (required, string)
+* `device_id`: the Device ID of the target device (required, string)
+* `credentials_secret`: the `credentials_secret` of the target device (required, string)
+* `interfaces`: the interfaces supported by the target device. Note that this should be an array of
+  objects, the whole interface definition must be present for every interface. Right now only
+  datastream interfaces are supported. (requires, array of objects)
+  
+This is an example of a `target_devices` object:
+
+```json
```
Comment (bettio) [inline guides/blocks/0119-virtual_device_pool.md:56]: sounds weird, but let's try using javascript

File: guides/blocks/0119-virtual_device_pool.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+`data` (and its `type`) must be compatible with the type declared in the mapping of the interface,
+so if an interface expects a `double` value but a `string` is received, the message will be
+discarded.
+
+If the interface supports an `explicit_timestamp`, the message `timestamp` is used as explicit
+timestamp.
+
+* `:key`: a unicode string that identifies the stream the message belongs to.
+* `:metadata`: additional message metadata as a map with string key and string value.
+* `:type`: message [data type](#data-and-types) (e.g. integer, real, boolean, etc...).
```
Comment (bettio) [inline guides/blocks/0119-virtual_device_pool.md:168]: I think that `[data type](#data-and-types)` is not a valid link

**Review by bettio — CHANGES_REQUESTED** [review]

File: guides/blocks/0119-virtual_device_pool.md (hunk 5 above)
Comment (rbino) [reply guides/blocks/0119-virtual_device_pool.md:168]: The whole block is actually a leftover, removing it

**Review by bettio — APPROVED** [review]

File: guides/blocks/0119-virtual_device_pool.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+Please refer to [Astarte documentation](https://docs.astarte-platform.org) to more information about
+Astarte concepts like Device IDs, Interfaces etc.
+
+## Target devices
+
+The block has two different behaviours regarding target devices, depending on its configuration:
+static and dynamic.
+
+With static devices, the set of target devices is known beforehand and all devices have to
+registered manually before using them inside Astarte Flow. Every device can have a different set of
```
Comment (atsetilam) [inline guides/blocks/0119-virtual_device_pool.md:22]: have to *be registered

**Pull request merged**